opal/threads/tsd.h

```diff
@@ -15,164 +15,6 @@
- */
-OPAL_DECLSPEC int opal_tsd_setspecific(opal_tsd_key_t key, void *value);
-
-
-/**
- * Get a thread-specific data value
- *
- * Get the data associated with the given key, as set by
- * opal_tsd_setspecific().  If opal_tsd_setspecific() hasn't been
- * called in the current thread with the given key, NULL is returned
- * in valuep.
- *
- * @param key[in]       Thread specific data key to modify
- * @param value[out]     Value to associate with key
- *
- * @retval OPAL_SUCCESS Success
- * @retval ENOMEM       Insufficient memory exists to associate the
- *                      value with the key
- * @retval EINVAL       Invalid key
- */
-OPAL_DECLSPEC int opal_tsd_getspecific(opal_tsd_key_t key, void **valuep);
-
-#else
-
-typedef pthread_key_t opal_tsd_key_t;
-
-static inline int
-opal_tsd_key_delete(opal_tsd_key_t key)
-{
-    return pthread_key_delete(key);
-}
-
-static inline int
-opal_tsd_setspecific(opal_tsd_key_t key, void *value)
-{
-    return pthread_setspecific(key, value);
-}
-
-static inline int
-opal_tsd_getspecific(opal_tsd_key_t key, void **valuep)
-{
-    *valuep = pthread_getspecific(key);
-    return OPAL_SUCCESS;
-}
-
-#endif
-
-/**
- * Create thread-specific data key
- *
- * Create a thread-specific data key visible to all threads in the
- * current process.  The returned key is valid in all threads,
- * although the values bound to the key by opal_tsd_setspecific() are
- * allocated on a per-thread basis and persist for the life of the
- * calling thread.
- *
- * Upon key creation, the value NULL is associated with the new key in
- * all active threads.  When a new thread is created, the value NULL
- * is associated with all defined keys in the new thread.
- *
- * The destructor parameter may be NULL.  At thread exit, if
- * destructor is non-NULL AND the thread has a non-NULL value
- * associated with the key, the function is called with the current
- * value as its argument.
- *
- * @param key[out]       The key for accessing thread-specific data
- * @param destructor[in] Cleanup function to call when a thread exits
- *
- * @retval OPAL_SUCCESS  Success
- * @retval EAGAIN        The system lacked the necessary resource to
- *                       create another thread specific data key
- * @retval ENOMEM        Insufficient memory exists to create the key
- */
-OPAL_DECLSPEC int opal_tsd_key_create(opal_tsd_key_t *key,
-                                      opal_tsd_destructor_t destructor);
-
-
-/**
- * Destruct all thread-specific data keys
- *
- * Destruct all thread-specific data keys and invoke the destructor
- *
- * This should only be invoked in the main thread.
- * This is made necessary since destructors are not invoked on the
- * keys of the main thread, since there is no such thing as
- * pthread_join(main_thread)
- *
- * @retval OPAL_SUCCESS  Success
- */
-OPAL_DECLSPEC int opal_tsd_keys_destruct(void);
-
-END_C_DECLS
-
-#endif /* OPAL_MTHREADS_TSD_H */
+#endif /* OPAL_THREADS_TSD_H */
```

Comment:
Ditto with `opal/threads/condition.h ` -- it looks like this file should be removed.

I'm going to stop marking the files in this directory: you should probably remove *all* files in `opal/threads` -- it doesn't make sense to leave old wrapper header files around.  If we're going to switch to be a `threads` framework, then 100% switch over to that (including the new location of header files).  Don't leave the old header files around that are just wrappers around the new header files.